bluemira/radiation_transport/flux_surfaces_maker.py

```diff
@@ -0,0 +1,303 @@
+        and their defaults.
+    equilibrium: Equilibrium
+        The equilibrium defining flux surfaces.
+    dx_mp: float (optional)
+        The midplane spatial resolution between flux surfaces [m]
+        (default: 0.001).
+    """
+
+    def __init__(self, equilibrium, dx_mp: float = 0.001):
+        self.eq = equilibrium
+        self.dx_mp = dx_mp
+
+        # Constructors
+        self.first_wall = None
+        self.flux_surfaces_ob_lfs = None
+        self.flux_surfaces_ob_hfs = None
+        self.flux_surfaces_ib_lfs = None
+        self.flux_surfaces_ib_hfs = None
+        self.x_sep_omp = None
+        self.x_sep_imp = None
+        self.result = None
+
+        # Pre-processing
+        o_points, _ = self.eq.get_OX_points()
+        self._o_point = o_points[0]
+        z = self._o_point.z
+        self._yz_plane = BluemiraPlane.from_3_points([0, 0, z], [1, 0, z], [1, 1, z])
+
+    @property
+    def flux_surfaces(self):
+        """
+        All flux surfaces in the ChargedParticleSolver.
+
+        Returns
+        -------
+        flux_surfaces: List[PartialOpenFluxSurface]
+        """
+        flux_surfaces = []
+        for group in [
+            self.flux_surfaces_ob_lfs,
+            self.flux_surfaces_ob_hfs,
+            self.flux_surfaces_ib_lfs,
+            self.flux_surfaces_ib_hfs,
+        ]:
+            if group:
+                flux_surfaces.extend(group)
+        return flux_surfaces
+
+    @staticmethod
+    def _process_first_wall(first_wall):
+        """
+        Force working first wall geometry to be closed and counter-clockwise.
+        """
+        first_wall = deepcopy(first_wall)
+
+        if not first_wall.check_ccw(axis=[0, 1, 0]):
+            bluemira_warn(
+                "First wall should be oriented counter-clockwise. Reversing it."
+            )
+            first_wall.reverse()
+
+        if not first_wall.closed:
+            bluemira_warn("First wall should be a closed geometry. Closing it.")
+            first_wall.close()
+        return first_wall
+
+    @staticmethod
+    def _get_array_x_mp(flux_surfaces):
+        """
+        Get x_mp array of flux surface values.
+        """
+        return np.array([fs.x_start for fs in flux_surfaces])
+
+    def _get_array_z_mp(flux_surfaces):
+        """
+        Get z_mp array of flux surface values.
+        """
+        return np.array([fs.z_start for fs in flux_surfaces])
+    
+    def _get_array_x_fw(flux_surfaces):
+        """
+        Get x_fw array of flux surface values.
+        """
+        return np.array([fs.x_end for fs in flux_surfaces])
+
+    def _get_array_z_fw(flux_surfaces):
+        """
+        Get z_fw array of flux surface values.
+        """
+        return np.array([fs.z_end for fs in flux_surfaces])
+    
+    def _get_array_alpha(flux_surfaces):
+        """
+        Get alpha angle array of flux surface values.
+        """
+        return np.array([fs.alpha for fs in flux_surfaces])
```

Comment:
All of these functions are currently unused. Also they all need the staticmethod decorator otherwise wont work.
I can see they could be useful at some point though.

Couldnt they all use `self.flux_surfaces` directly and therefore not need to be staticmethods or private methods? 

eg:
```python
    def get_array_alpha(self):
        """
        Get alpha angle array of flux surface values.
        """
        return np.array([fs.alpha for fs in self.flux_surfaces])
```

the same is true of `_get_array_x_mp` but it is used currently